addons/escoria-core/game/core-scripts/esc_controller.gd

```diff
@@ -0,0 +1,244 @@
+# This script contains functions called by gamedev's game.gd.
+# These functions convert input actions into game actions. For example:
+# "click on background" -> player walks to position
+# "click on item" -> player walks to item then performs the action defined
+# by current verb
+
+class_name ESCController
+
+# Makes an object walk to a destination. This can be either a 2D position or
+# another object.
+#
+# #### Parameters
+#
+# - moving_obj_id: global id of the object that needs to move
+# - destination: Position2D or String of the object the moving object has head
+# to
+# - is_fast: if true, the walk is performed at fast speed (defined in the moving
+# object.
+func perform_walk(moving_obj_id: String, destination, is_fast: bool = false):
```

Comment:
Ah, I had done that in the first place then reverted because I needed both the id AND the ESCObject. But after multiple changes, I didn't see that I could set this back again. Thanks!